index.js

```diff
@@ -8,6 +8,78 @@ function isString(s) {
   return 'string' === typeof s
 }
 
+var hasLink = exports.hasLink =
```

Comment:
I'm ok with these functions returning an exec's match results, but if they are going to do that, they should be named something like `extractLink` instead of `hasLink`. Perhaps the plural version, `extractLinks`, will iterate all matches and put them into an array.